android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -57,25 +62,34 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        handleVisibility();
-
-        globalSearchAdapter = new GlobalSearchAdapter(results, this);
-        searchRecyclerView.setAdapter(globalSearchAdapter);
-        searchRecyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
-        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+        setUpRecyclerView();
 
         if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
             searchText = savedInstanceState.getString(SEARCH);
```

Comment:
Remove this now, you don't need to save the searchText in the ```saveInstanceState```. You can save it inside the ViewModel which will outlast the life of this Activity.